plugins/lsp/lspsaga.nix

```diff
@@ -7,220 +7,651 @@
+            When have `relatedInformation`, diagnostic message is extended to show it.
+          '';
+
+          diagnosticOnlyCurrent = helpers.defaultNullOpts.mkBool false ''
+            Only show diagnostic virtual text on the current line.
+          '';
+
+          keys = {
+            execAction = mkKeymapOption "o" "Execute action (in jump window).";
+
+            quit = mkKeymapOption "q" "Quit key for the jump window.";
+
+            toggleOrJump = mkKeymapOption "<CR>" ''
+              Toggle or jump to position when in `diagnostic_show` window.
+            '';
+
+            quitInShow = mkKeymapOption ["q" "<ESC>"] ''
+              Quit key for the diagnostic_show window.
+            '';
+          };
         };
 
-        warning = mkOption {
-          type = types.nullOr types.str;
-          default = null;
-          description = "Warning diagnostic header";
+        codeAction = {
+          numShortcut = helpers.defaultNullOpts.mkBool true ''
+            Whether number shortcut for code actions are enabled.
+          '';
+
+          showServerName = helpers.defaultNullOpts.mkBool false ''
+            Show language server name.
+          '';
+
+          extendGitSigns = helpers.defaultNullOpts.mkBool false ''
+            Extend gitsigns plugin diff action.
+          '';
+
+          onlyInCursor = helpers.defaultNullOpts.mkBool true "";
+
+          maxHeight = helpers.defaultNullOpts.mkNullable (types.numbers.between 0.0 1.0) "0.3" "";
+
+          keys = {
+            quit = mkKeymapOption "q" "Quit the float window.";
+
+            exec = mkKeymapOption "<CR>" "Execute action.";
+          };
         };
 
-        hint = mkOption {
-          type = types.nullOr types.str;
-          default = null;
-          description = "Hint diagnostic header";
+        lightbulb = {
+          enable = helpers.defaultNullOpts.mkBool true "Enable lightbulb.";
+
+          sign = helpers.defaultNullOpts.mkBool true "Show sign in status column.";
+
+          debounce = helpers.defaultNullOpts.mkInt 10 "Timer debounce.";
+
+          signPriority = helpers.defaultNullOpts.mkInt 40 "Sign priority.";
+
+          virtualText = helpers.defaultNullOpts.mkBool true "Show virtual text at the end of line.";
+
+          enableInInsert = helpers.defaultNullOpts.mkBool true "Enable in insert.";
         };
 
-        info = mkOption {
-          type = types.nullOr types.str;
-          default = "  Info";
-          description = "Info diagnostic header";
+        scrollPreview = {
+          scrollDown = mkKeymapOption "<C-f>" "Scroll down.";
+
+          scrollUp = mkKeymapOption "<C-b>" "Scroll up.";
         };
-      };
 
-      maxDialogWidth = mkOption {
-        type = types.nullOr types.int;
-        default = null;
-        description = "Maximum dialog width";
-      };
+        requestTimeout = helpers.defaultNullOpts.mkInt 2000 "Request timeout.";
```

Comment:
This does not appear to be documented too

Reply:
removed